tests/api/parser.py

```diff
@@ -0,0 +1,40 @@
+
+# ################ MESH ################
+
+
+# TODO: this and the following function should return a list of values and not issue only the first object
+def parse_layer_hash_from_layer_query_res(layer_query_response):
+    # {..., 'layer': [{'hash': '[\d\w=]+', ...}]}
+    return layer_query_response['layer'][0]['hash']
```

Comment:
what should happen if the value is not found? just throw an exception?

Reply:
```suggestion
    return [h['hash'] for h in layer_query_response['layer']]
```

This should take care of the TODO, while also not raising an exception if the list in `layer` is empty (just returns an empty list in that case).

It still raises an exception if `layer` is missing or a value in the list is missing the `hash` key. That seems like reasonable behaviour to me, but if you think it shouldn't raise an exception in that case, you can replace the `['layer']` with `.get('layer', [])` and/or the `['hash']` with `.get('hash')`.

Reply:
You can use the same list comprehension in the following methods, too.

Reply:
I need only 1 value in this case, the name of the function implies one value as well.
next I will add a multiple value return function that will do what you suggested here.